signing/signer.go

```diff
@@ -0,0 +1,58 @@
+package signing
+
+import (
+	"errors"
+	"github.com/spacemeshos/ed25519"
+	"github.com/spacemeshos/go-spacemesh/log"
+)
+
+type BlsFetcher interface {
+	FetchBlsKey(edKey []byte) []byte
+}
+
+type ActivationValidator interface {
+	IsActive(blsKey []byte) bool
+}
+
+type Signer interface {
+	Sign(m []byte) []byte
+	Verifier() Verifier
+}
+
+type EdSigner struct {
+	privKey ed25519.PrivateKey // the pub & private key
+	pubKey  ed25519.PublicKey  // only the pub key part
+}
+
+func NewEdSignerFromBuffer(buff []byte) (*EdSigner, error) {
+	if len(buff) < 32 {
+		return nil, errors.New("buffer too small")
+	}
+	return &EdSigner{privKey: buff, pubKey: buff[:32]}, nil
+}
+
+func NewEdSigner() *EdSigner {
+	pub, priv, err := ed25519.GenerateKey(nil)
```

Comment:
add a log print with the public key (without the private key..)

Reply:
Added

Reply:
Try again

Reply:
Added where we call NewEdSigner